Frontend/Pages/Projects/LegalRequirements/FoundationConsent.cshtml.cs

```diff
@@ -0,0 +1,93 @@
+using Data;
+using Frontend.Models.Forms;
+using Frontend.Models.LegalRequirements;
+using Frontend.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Frontend.ExtensionMethods;
+
+namespace Frontend.Pages.Projects.LegalRequirements
+{
+    public class FoundationConsentModel : CommonPageModel
+    {
+        private readonly IProjects _projects;
+
+        public FoundationConsentModel(IProjects projects)
+        {
+            _projects = projects;
+        }
+
+        [BindProperty] public FoundationConsentViewModel FoundationConsentViewModel { get; set; } = new FoundationConsentViewModel();
+
+        public IList<RadioButtonViewModel> RadioButtonsYesNoNotApplicable { get; set; }
+
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            var project = await _projects.GetByUrn(Urn);
+            IncomingTrustName = project.Result.IncomingTrustName;
+
+            RadioButtonsYesNoNotApplicable = GetRadioButtons(project.Result.LegalRequirements.FoundationConsent.ToDescription());
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            var project = await _projects.GetByUrn(Urn);
+
+            if (ModelState.IsValid is false)
+            {
+                RadioButtonsYesNoNotApplicable = GetRadioButtons(project.Result.LegalRequirements.FoundationConsent.ToDescription());
+                return Page();
+            }
+            project.Result.LegalRequirements.FoundationConsent = FoundationConsentViewModel.FoundationConsent;
+            await _projects.Update(project.Result);
+            if (ReturnToPreview)
+            {
+                return RedirectToPage(Links.HeadteacherBoard.Preview.PageName, new { Urn });
+            }
+
+            return RedirectToPage("/Projects/LegalRequirements/Index", new { Urn });
+        }
+
+        private IList<RadioButtonViewModel> GetRadioButtons(string valueSelected)
```

Comment:
This could be moved into a base class.

Reply:
Or maybe the viewmodel would be better?

Reply:
Viewmodel is a good idea, have created a common viewmodel for them and moved the logic there.